src/proto/h1/role.rs

```diff
@@ -205,8 +198,8 @@ impl Http1Transaction for Server {
         headers.reserve(headers_len);
 
         for header in &headers_indices[..headers_len] {
-            let name = header_name!(&slice[header.name.0..header.name.1]);
-            let value = header_value!(slice.slice(header.value.0..header.value.1));
+            let name = header_name!(&slice[header.name.0..header.name.1])?;
```

Comment:
I think the macros could be written to just `return Err(whatever)` inside them, and then these lines don't really need to change, ya?

Reply:
Not sure i follow, `header_name` and `header_value` macros returns a `Result`, from the new `maybe_panic` macro.

Reply:
In the macro, you make it so it doesn't wrap the value back in `Ok`, and on error `return Err(err)`. Then at the use site, we don't need to add a question mark.